### PR TITLE
Add support for SPDX license headers

### PR DESCRIPTION
**Pull request by craigez**
Just a heads up I'm working on this to align the set of licenses with LiD. There's a few things that need thought/resolution:

- "There is no standard license header for the license"
- How licenses.py is handled and forcing the '-header' into the map later

### CODE REVIEW

**Comment by craigez** [comment]
Related to #2

**Comment by garious** [comment]
Nice! Great contribution. I'm out of town right now but can review early next week. Or let me know if I should hold off until you remove the WIP.

File: codebom/licenseidentifier.py [hunk 1]
```diff
@@ -77,7 +81,9 @@ def _make_template_words(id):
     return text.split()
 
 def _make_template_words_map(license_ids):
-    return {id: _make_template_words(id) for id in license_ids if os.path.isfile(_template_path(id))}
+    word_map = {id: _make_template_words(id) for id in license_ids if os.path.isfile(_template_path(id))}
```
Comment (garious) [inline codebom/licenseidentifier.py]: How about?

```python
ids = license_ids + [id + "-header" for id in license_ids]
return {id: _make_template_words(id) for id in ids if os.path.isfile(_template_path(id))}
```

File: update_licenses.py [hunk 2]
```diff
@@ -28,14 +28,20 @@ def license_text_to_str(literal):
     return xml_to_str(literal.value.firstChild) + '\n'
 
 def get_license_text(licenseId):
+    return get_license_term(licenseId, "http://spdx.org/rdf/terms#licenseText")
```
Comment (garious) [inline update_licenses.py]: Can you snake_case all the local variables?

File: codebom/licenseidentifier.py [hunk 3]
```diff
@@ -51,6 +51,8 @@ def _identify_license_text(license_words, template_words_map):
     def _get_score(id):
         matched_len, templ_len = ngram_matches.get(id)
         # Note: For Python 2.7, either operand must be of type float.
+        if templ_len == 0:
```
Comment (garious) [inline codebom/licenseidentifier.py]: Can you put this check above the Python 2.7 note?

Comment (craigez) [inline codebom/licenseidentifier.py]: I think this is one of the areas I'll update once I worked out what to do with the headers that contain a message saying there is no header.

File: update_licenses.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 def get_license_ids():
     graph = rdflib.Graph()
     graph.parse('http://spdx.org/licenses/index.html', 'rdfa')
-    ref = rdflib.URIRef('http://spdx.org/rdf/terms#licenseId')
+    ref = rdflib.URIRef('http://spdx.org/rdf/terms#license_id')
```
Comment (garious) [inline update_licenses.py]: Is that right?

Comment (craigez) [inline update_licenses.py]: Argh!

File: update_licenses.py [hunk 5]
```diff
@@ -16,15 +16,26 @@ def get_license_ids():
     return map(lambda x: x[1].value, objs)
 
 def lit_to_str(lit):
-    s = lit.nodeValue.encode('utf-8') if lit.nodeValue else ''
+    try:
```
Comment (garious) [inline update_licenses.py]: Is there a way to LBYL here?

Comment (craigez) [inline update_licenses.py]: You want an explicit version check? I thought it best to "feature" check, though agree it's not pretty. 

(I actually broke python27 somewhere in this process, so working on a fix for that)

Comment (garious) [inline update_licenses.py]: Sorry for not mentioning this earlier, but I have no need for Python27 support. Feel free to delete any code that references it. The tox.ini only mentions py35 and py36, which represents what I use and what I want to use, respectively.

Comment (garious) [inline update_licenses.py]: https://github.com/codebom/codebom/commit/5e8db2592bf9b6dcafb6386210d078dc66afd9ca

**Comment by craigez** [comment]
Okay, but then the update_licenses script doesn't work with Python3 as is in master. I'll just make it Python 3 compatible only then.

**Comment by garious** [comment]
Awesome, thanks. update_licenses being Python 2 is an oversight. It's ignored by the test suite. I'd like to integrate it and here's the ticket to do it: https://github.com/codebom/codebom/issues/5

**Comment by craigez** [comment]
Think I'm good for you to finish reviewing this now.

**Comment by garious** [comment]
The patches look good. Have you looked into how introducing headers affects `codebom scan`? Seems like headers ought to be ignored when a LICENSE file with the full license text is present. Have you looked much at this code: https://github.com/codebom/codebom/blob/master/codebom/scan.py#L46

**Comment by craigez** [comment]
I haven't looked into it in detail. I'll play with some examples. Perhaps there will be a need to add an example into the codebom repository.

**Comment by garious** [comment]
Take a look at this test case:
https://github.com/codebom/codebom/blob/master/codebom/scan_test.py#L143

If a.txt had license 'MIT-header' instead of 'MIT', I'd expect to see license 'MIT' in the coalesced output.

And what do you think about the text 'MIT-header' ever making it into a BOM? Should that just be 'MIT' and the end user can assume a file containing the header is equivalent to one containing the full license? It's easy to imagine it being standard practice to add headers to source code and those headers reference some URL (not a LICENSE file) with the full license text.

**Comment by garious** [comment]
@craigez, want me to merge as-is? I can polish up the coalescing in the near future.

**Comment by craigez** [comment]
I just haven't had a chance to take a look, I plan to though as it is annoying to have a mismatch in this case. I need to spend a little time to better understand when CodeBOM scans source files to decide exactly how to resolve this.

**Comment by craigez** [comment]
I totally dropped the ball on this. Happy for you to merge and tidy up coalescing later.

**Pull request merged**